### ISSUE TITLE
Miner panic: minner ranks not computed yet

### ISSUE THREAD

**Issue by peterlimg** [0chain/0chain#862]
This happens on `fix/state-hash-mismatch` branch, but all branches should have the issue.

```go
2022-01-06T18:38:39.384Z        DPANIC  round/entity.go:590     miner ranks not computed yet: 0, random seed: 0, round: 7615
0chain.net/chaincore/round.(*Round).GetMinerRank
        /0chain/go/0chain.net/chaincore/round/entity.go:590
0chain.net/miner.(*Chain).TryProposeBlock
        /0chain/go/0chain.net/miner/protocol_round.go:296
0chain.net/miner.(*Chain).AddVRFShare
        /0chain/go/0chain.net/miner/protocol_bls.go:301
0chain.net/miner.(*Chain).addMyVRFShare
        /0chain/go/0chain.net/miner/protocol_round.go:91
0chain.net/miner.(*Chain).StartNextRound
        /0chain/go/0chain.net/miner/protocol_round.go:207
0chain.net/miner.(*Chain).moveToNextRoundNotAhead
        /0chain/go/0chain.net/miner/protocol_round.go:1133
(END)
```

Miner tried to compute the ranks when VRF shares reached threshold, however it could return early before it starts the miner ranks computation if it mistakenly thought the round already has the seed.  Check the code below,

```
func (r *Round) SetRandomSeed(seed int64, minersNum int) {
	if atomic.LoadUint32(&r.hasRandomSeed) == 1 {
		return
	}

	r.mutex.Lock()
	r.minerPerm = computeMinerRanks(seed, minersNum)
	r.mutex.Unlock()

	r.setRandomSeed(seed)
	r.setState(RoundVRFComplete)
}
```

The `r.hasRandomSeed` could be set to `1` in `SetRandomSeedForNotarizedBlock`, so if it is marked as has seed, but not yet finish the ranks computation, we will have this panic.

**Comment by peterlimg** [comment]
I'll send a fix to the `fix/state-hash-mismatch`, there are fix codes were made on top of it.

**Comment by peterlimg** [comment]
Fixed by 53b3e3c32fcefc1d53309bf5f954a25186b8102f
